std/algorithm/searching.d

```diff
@@ -119,11 +119,15 @@ template all(alias pred = "a")
     Performs (at most) $(BIGOH range.length) evaluations of `pred`.
      +/
     bool all(Range)(Range range)
-    if (isInputRange!Range &&
-        (__traits(isTemplate, pred) || is(typeof(unaryFun!pred(range.front)))))
     {
         import std.functional : not;
 
+        static assert(isInputRange!Range, Range.stringof, " must be an"
+                , " InputRange");
+        static assert(__traits(isTemplate, pred)
+                    || is(typeof(unaryFun!pred(range.front)))
+                , "The passed predicate (pred), must be usable by unaryFun");
```

Comment:
Maybe this as a compromise:
```suggestion
                    || is(typeof(unaryFun!pred(noreturn.init)))
                , "The passed predicate (pred) does not accept an argument");
```
That checks that pred is callable but does not hide a parameter type error, which is more useful to see.